samza-core/src/main/java/org/apache/samza/container/SamzaContainerListener.java

```diff
@@ -23,6 +23,11 @@
  */
 public interface SamzaContainerListener {
 
+  /**
+   * Method invoked before the {@link org.apache.samza.container.SamzaContainer} is started
+   */
+  void beforeStart();
+
   /**
    *  Method invoked when the {@link org.apache.samza.container.SamzaContainer} has successfully transitioned to
```

Comment:
Renamed the method names. Still prefer to keep this interface separately from ProcessorLifecycleListener, since this interface should serve a different purpose in interaction between the StreamProcessor and the SamzaContainer/JobCoordinator.